e2e/testsuite/testsuite.go

```diff
@@ -0,0 +1,286 @@
+package testsuite
+
+import (
+	"context"
+	"fmt"
+	"io/ioutil"
+	"strings"
+	"testing"
+	"time"
+
+	dockerclient "github.com/docker/docker/client"
+	"github.com/strangelove-ventures/ibctest"
+	"github.com/strangelove-ventures/ibctest/chain/cosmos"
+	"github.com/strangelove-ventures/ibctest/ibc"
+	"github.com/strangelove-ventures/ibctest/relayer"
+	"github.com/strangelove-ventures/ibctest/testreporter"
+	"github.com/stretchr/testify/suite"
+	"go.uber.org/zap"
+	"go.uber.org/zap/zaptest"
+
+	"github.com/cosmos/ibc-go/v4/e2e/testconfig"
+)
+
+const (
+	SourceRelayerName      = "srcRly"
+	DestinationRelayerName = "dstRly"
+)
+
+// E2ETestSuite has methods and functionality which can be shared among all test suites.
+type E2ETestSuite struct {
+	suite.Suite
+	chainPairs     map[string]chainPair
+	logger         *zap.Logger
+	Client         *dockerclient.Client
+	network        string
+	startRelayerFn func(relayer ibc.Relayer)
+}
+
+// chainPair is a pairing of two chains which will be used in a test.
+type chainPair struct {
+	srcChain, dstChain *cosmos.CosmosChain
```

Comment:
I presume there is a difficulty with making this ibc.Chain?

Reply:
In certain places we depend on the concrete type `*cosmos.CosmosChain`. Not in this PR but in the follow up tests. I'll make a note of this to verify that this can't just be `ibc.Chain`.